Observer/QuoteSubmitSuccessObserver.php

```diff
@@ -0,0 +1,137 @@
+     * @param AvaTaxLogger                $logger
+     */
+    public function __construct(
+        AssociatedTaxableFactory $associatedTaxableInterfaceFactory,
+        AssociatedTaxableRepository $associatedTaxableRepository,
+        AvaTaxLogger $logger,
+        \ClassyLlama\AvaTax\Helper\AssociatedTaxable $associatedTaxableHelper
+    ) {
+        $this->associatedTaxableFactory = $associatedTaxableInterfaceFactory;
+        $this->associatedTaxableRepository = $associatedTaxableRepository;
+        $this->avataxLogger = $logger;
+        $this->associatedTaxableHelper = $associatedTaxableHelper;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var Quote $quote */
+        $quote = $observer->getQuote();
+        /** @var Order $order */
+        $order = $observer->getOrder();
+
+        $quoteItemIdOrderItemMap = [];
+        foreach ($order->getItems() as $orderItem) {
+            $quoteItemIdOrderItemMap[$orderItem->getQuoteItemId()] = $orderItem;
+        }
+        /** @var Quote\Item $quoteItem */
+        foreach ($quote->getAllItems() as $quoteItem) {
+            // item-level associated taxables
+            $associatedTaxables = $quoteItem->getAssociatedTaxables();
+            if ($associatedTaxables !== null) {
+                foreach ($associatedTaxables as $associatedTaxableData) {
+                    $associatedTaxableData = $this->associatedTaxableHelper->updateData(
+                        $associatedTaxableData,
+                        $quoteItem->getStoreId()
+                    );
+                    /** @var AssociatedTaxableInterface $associatedTaxable */
+                    $associatedTaxable = $this->associatedTaxableFactory->create(
+                        [
+                            'data' => $associatedTaxableData
+                        ]
+                    );
+                    $associatedTaxable->setInvoiceId(null);
+                    if (isset($quoteItemIdOrderItemMap[$quoteItem->getId()])) {
+                        /** @var Order\Item $orderItem */
+                        $orderItem = $quoteItemIdOrderItemMap[$quoteItem->getId()];
+                        $associatedTaxable->setOrderItemId((int)$orderItem->getId());
+                        $associatedTaxable->setOrderId($order->getId());
+                        try {
+                            $this->associatedTaxableRepository->save($associatedTaxable);
+                        } catch (\Exception $e) {
+                            $this->avataxLogger->error(
+                                "There was a problem saving associated taxable item information for order {$order->getId()}"
+                            );
+                        }
+                    }
+                }
+            }
+        }
+
+        if (!$quote->isVirtual()) {
+            $shippingAddress = $quote->getShippingAddress();
```

Comment:
Probably need to refactor this to use `\Magento\Quote\Model\Quote::getAllShippingAddresses` to support multiple addresses.